### ISSUE TITLE
Creating a thread on a native closure results in segfault

### ISSUE THREAD

**Issue by kyleedwardsny** [albertodemichelis/squirrel#186]
```
local coro = ::newthread(some_native_closure);
```

Ideally, this should Do The Right Thing and call the native closure with the arguments given to `call()`. If this can't be implemented, it should at the very least gracefully throw an exception rather than segfaulting.

**Comment by albertodemichelis** [comment]
Yes indeed, base_newthread() should check for the closure type. I'll fix that.
